### ISSUE TITLE
/api/v1/series: no matches if no time constraints provided

### ISSUE THREAD

**Issue by vthriller** [VictoriaMetrics/VictoriaMetrics#91]
```
$ curl -g 'localhost:8428/api/v1/series?match[]={__name__=~"foo.*"}'; echo
{"status":"success","data":[]}

$ curl -g 'localhost:8428/api/v1/series?match[]={__name__=~"foo.*"}&start=0'; echo
{"status":"success","data":[{"__name__":"foo.bar.baz"},…]}
```

Here's how Prometheus acts:

```
$ curl -g 'localhost:9090/api/v1/series?match[]={__name__=~"foo.*"}'; echo
{"status":"success","data":[{"__name__":"foo.bar.baz"},…]}

$ curl -g 'localhost:9090/api/v1/series?match[]={__name__=~"foo.*"}&start='$(date +%s -d '1 hour ago'); echo
{"status":"success","data":[]}
```

**Comment by vthriller** [comment]
https://github.com/prometheus/prometheus/blob/fef150f1b5e48652ec6779e2129ae9a0cf13db8a/web/api/v1/api.go#L458-L466

**Comment by hagen1778** [comment]
Hi @vthriller!

Thank you for the question! 

VictoriaMetrics set default value for `start` param as 5min ago - see https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/app/vmselect/prometheus/prometheus.go#L312
 Please, can you share your use case and why it is a problem for you?

**Comment by vthriller** [comment]
> Please, can you share your use case and why it is a problem for you?

This issue appears to be the reason why https://github.com/go-graphite/carbonapi can't show any metrics when VM is used as a backend for it.

I obviously can't tell whether this endpoint is used in this way anywhere else, but at the very least, if not fixed, it's a good idea to document this incompatibility, hence this issue.

**Comment by vthriller** [comment]
> carbon can't show any metrics

well, at least in my case, as I'm currently ingesting tons of archival data into VM, thus I don't have anything for at least past 1h.

**Comment by valyala** [comment]
@vthriller , could you verify the issue is fixed in 62a735347963f0522c3209145e95e7594da9e5cb ?

**Comment by vthriller** [comment]
> could you verify whether the issue is fixed in 62a7353 ?

It definitely is; thanks!

**Comment by valyala** [comment]
The fix is available in [v1.22.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).

**Comment by valyala** [comment]
@vthriller , just FYI, starting from [v1.24.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases) `/api/v1/series` requires explicit `start` argument set to `0` or other small value in order to get time series info over all the available time range. By default `start` is set 5 minutes before `end` in order to reduce potential overload on big time series database when `start` isn't set.
